File: jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/event/UXKeyEvent.php
<?php
namespace php\gui\event;

/**
 * Class UXKeyEvent
 * @package php\gui\event
 */
class UXKeyEvent extends UXEvent
{
    /**
     * @var string
     */
    public $character;

    /**
     * @var string
     */
    public $text;

    /**
     * @var string
     */
    public $codeName;

    /**
     * @var bool
     */
    public $altDown;

    /**
     * @var bool
     */
    public $controlDown;

    /**
     * @var bool
     */
    public $shiftDown;

    /**
     * @var bool
     */
    public $metaDown;

    /**
     * @var bool
     */
    public $shortcutDown;
}

File: jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/event/UXKeyEvent.java
package org.develnext.jphp.ext.javafx.classes.event;

import javafx.scene.input.KeyEvent;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Getter;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "event\\UXKeyEvent")
public class UXKeyEvent extends UXEvent {
    interface WrappedInterface {
        @Property String character();
        @Property String text();

        @Property boolean shiftDown();
        @Property boolean controlDown();
        @Property boolean altDown();
        @Property boolean metaDown();
        @Property boolean shortcutDown();
    }

    public UXKeyEvent(Environment env, KeyEvent wrappedObject) {
        super(env, wrappedObject);
    }

    public UXKeyEvent(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Reflection.Signature
    public void __construct(KeyEvent event, Object sender) {
        __wrappedObject = new KeyEvent(sender, null, event.getEventType(), event.getCharacter(), event.getText(), event.getCode(), event.isShiftDown(), event.isControlDown(), event.isAltDown(), event.isMetaDown());
    }

    @Override
    public KeyEvent getWrappedObject() {
        return (KeyEvent) super.getWrappedObject();
    }

    @Getter
    public String getCodeName() {
        return getWrappedObject().getCode().getName();
    }
}
